networkx/generators/time_series.py

```diff
@@ -0,0 +1,72 @@
+"""
+Time Series Graphs
+"""
+import itertools
+
+import networkx as nx
+
+__all__ = ["visibility_graph"]
+
+
+def visibility_graph(series):
```

Comment:
I did a quick google of the term "visibility graph" and the first hit was [a wiki article](https://en.wikipedia.org/wiki/Visibility_graph) which gave me pause about this name. According to the article, the term is used in robotics to capture the same concept, but in spatial dimensions (i.e. 2D or 3D) instead of the "time series" (i.e. 1D) interpretation used here.

Perhaps we should consider a different, more specific name for this graph generation function then? Maybe `sequential_visibility_graph` or `series_visibility_graph`? I don't feel *too* strongly about it, but it's worth thinking about naming now!

Reply:
So perhaps the grouping "Time Series" is what we should change?

How about "Scalar Fields"? You can think of time series as a 1D Scalar field.

This can then include the another PR I made  #6919 - which is about generating Gradient Network of a scalar field substrate graph.

Reply:
> From what perspective are you finding these algorithms and being interested enough to submit PRs? 

First of all I find Graph Theory generally interesting. Specifically at work we modeled data into graphs with two immediate main aims: 1) standard data structures to hold data from disparate sources. 2) investigate those data structures and create larger graphs by finding connections. But in the long run also to gain insight by utilizing Graph Theory to investigate the data gathered.

I was looking into data source  that had time series characteristics - and was looking for a way to model that as a graph. Visibility Graphs looked liked the way to go. 

Why do a PR? For quite a while I wanted to get into open source. I used NetworkX, and I think its an excellent lib. So I thought why not contribute back.

> What training/literature are you coming from?

I always loved Math and Physics (and science in general), I still do. I did my BSc in Physics/Math. Some Master degree courses. My plan to go all way. PhD, Academia, Research. The whole thing. But then life happened. 
Started working in software ~30 years ago. Gained experience in many things. Now I mainly do backend engineering. What I like most is to have mix of software engineering with the science/math. When there's a combination of both, and things go well, that's when you get to make "innovative" things, sometimes even "magical" things. You can check my [linkedin](https://www.linkedin.com/in/eranrivlis/) for more.

Reply:
Thanks @dschult for digging into the background and the nice summary!

> Perhaps of most import is that both "time series" and "visibility graph" seem to be standard terms for this style of analysis. I don't find very much using "scalar field" (perhaps too broad a term -- or maybe just not as catchy -- or maybe I just didn't search well)

I'm convinced! +1 for sticking with the original naming convention given it's better alignment with the general landscape.

Reply:
Thanks for that summary of your background. It's very nice to have you here and to have your help. :)